File: mycollab-core/src/main/java/com/esofthead/mycollab/core/arguments/DateTimeSearchField.java
package com.esofthead.mycollab.core.arguments;

import java.util.Date;

public class DateTimeSearchField extends SearchField {

	public static String LESSTHAN = "<";
	public static String LESSTHANEQUAL = "<=";
	public static String GREATERTHAN = ">";
	public static String GREATERTHANEQUAL = ">=";
	public static String EQUAL = "=";
	public static String NOTEQUAL = "<>";

	private Date value;
	private String comparision;

	public DateTimeSearchField() {
		this(AND, null);
	}

	public DateTimeSearchField(String oper, Date value) {
		this(oper, DateTimeSearchField.LESSTHAN, value);
	}

	public DateTimeSearchField(String oper, String comparision, Date value) {
		this.operation = oper;
		this.value = value;
		this.comparision = comparision;
	}

	public Date getValue() {
		return value;
	}

	public void setValue(Date value) {
		this.value = value;
	}

	public String getComparision() {
		return comparision;
	}

	public void setComparision(String comparision) {
		this.comparision = comparision;
	}

}

File: mycollab-dao/src/main/java/com/mycollab/db/arguments/DateTimeSearchField.kt
package com.mycollab.db.arguments

import com.mycollab.core.utils.DateTimeUtils
import java.util.*

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class DateTimeSearchField(operation: String, comparison: String, dateVal: Date, value: Date = DateTimeUtils.convertDateTimeToUTC(dateVal)) : SearchField(operation) {
    companion object {
        val LESS_THAN = "<"
        val LESS_THAN_EQUAL = "<="
        val GREATER_THAN = ">"
        val GREATER_THAN_EQUAL = ">="
        val EQUAL = "="
        val NOT_EQUAL = "<>"
    }
}